test/controllers/confirmation.controller.spec.unit.ts

```diff
@@ -163,6 +164,154 @@ describe("Company no longer required confirmation screen tests", () => {
     expect(resp.text).toContain("filed by 10 March 2028");
   });
 
+  it("should render the not eligible page (no open compliance case) when reason code is NO_OPEN_COMPLIANCE_CASE",
+    async () => {
+
+    mockCacheService.mockClear();
+    mockPTFSession.mockClear();
+    mockActiveFeature.mockClear();
+    mockCallProcessorApi.mockRestore();
+    loadCompanyAuthenticatedSession(mockCacheService, COMPANY_NUMBER, EMAIL);
+    const dummyProfile = getDummyCompanyProfile(true, true);
+    mockPTFSession.mockImplementationOnce(() => dummyProfile);
+    mockPTFSession.mockImplementationOnce(() => true);
+    mockActiveFeature.mockImplementationOnce(() => true);
+
+    mockCallProcessorApi.prototype.constructor.mockImplementation(() => Promise.resolve((
+      {
+        data: {
+          filing_due_on: "null",
```

Comment:
According to the spec, filing_due_on field isn't even present in the JSON for a 400 response. So I'm interested in what the API does actually return? Did/could you try removing the "null" value here (and does test still pass)?

Reply:
They pass without the filing_due_on field. I'll remove them 👍